### PR TITLE
Process preview in target frame

### PR DESCRIPTION
**Pull request by K1llMan**
Preview frame processing in targets frame.

### CODE REVIEW

**Comment by henryruhs** [comment]
Thanks for this addition, can you please merge against the next branch as we merge into that one.

File: run.py [hunk 1]
```diff
@@ -273,7 +281,7 @@ def start():
     frames_checkbox.place(x=60,y=450,width=240,height=31)
 
     # Start button
-    start_button = tk.Button(window, text="Start", bg="#f1c40f", relief="flat", borderwidth=0, highlightthickness=0, command=lambda: [save_file(), start()])
+    start_button = tk.Button(window, text="Start", bg="#f1c40f", relief="flat", borderwidth=0, highlightthickness=0, command=lambda: [save_file(), start_thread()])
```
Comment (henryruhs) [inline run.py:284]: Keep it start() but fire up a preview_thread() there

File: run.py [hunk 2]
```diff
@@ -228,6 +232,10 @@ def start():
     status("swap successful!")
 
 
+def start_thread():
```
Comment (henryruhs) [inline run.py:235]: Rename to preview thread

File: core/swapper.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ def process_faces(source_face, frame, progress, all_faces=False):
     return frame
 
 
-def process_video(source_img, frame_paths):
+def process_video(source_img, frame_paths, process_frame):
```
Comment (henryruhs) [inline core/swapper.py:44]: process_frame is a callback that you fire? lets call it that way then... preview_callback

File: run.py [hunk 4]
```diff
@@ -116,6 +116,17 @@ def preview_image(image_path):
     img_label.pack()
 
 
+def preview_frame(frame):
```
Comment (henryruhs) [inline run.py:119]: update_preview()

**Review by henryruhs — CHANGES_REQUESTED** [review]

**Comment by henryruhs** [comment]
Can you check if the multi UI problem is fixed?

**Comment by henryruhs** [comment]
Please resolve the conflicts

**Comment by K1llMan** [comment]
I'll make new PR with separate window from 'next' branch.

**Comment by henryruhs** [comment]
Looking forward for the final version.

**Pull request closed without merging**